src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java

```diff
@@ -84,6 +87,14 @@ public static StructuredSyslogMessage fromString(
 
     private void  deserialize(final String stringMessage) {
 
+        // Check if the RFC5424 MSGID and STRUCTURED-DATA fields are empty.
+        // This avoids throwing an exception and also strips the "- - " from the message.
+        // See: https://github.com/Graylog2/graylog2-server/issues/1161
+        if (stringMessage.startsWith(EMPTY_STRUCTURED_DATA_PREFIX)) {

```

Comment:
No need to excuse, we merged that pretty quickly. :wink: Thank you for still looking at it!

At this point we know that this is a RFC5424 message and so there has to be a `MSGID` and `STRUCTURED-DATA` section. Since both can be `NIL` (`-`), we quickly check for that.

At least that is my understanding. Did I miss something?